src/app/util/ember-compatibility-functions.cpp

```diff
@@ -479,6 +479,22 @@ CHIP_ERROR WriteSingleClusterData(ClusterInfo & aClusterInfo, TLV::TLVReader & a
     attributePathParams.mFieldId    = aClusterInfo.mFieldId;
     attributePathParams.mFlags.Set(AttributePathParams::Flags::kFieldIdValid);
 
+    // TODO: Refactor WriteSingleClusterData and all dependent functions to take ConcreteAttributePath instead of ClusterInfo
+    // as the input argument.
+    AttributeAccessInterface * attrOverride = findAttributeAccessOverride(aClusterInfo.mEndpointId, aClusterInfo.mClusterId);
+    if (attrOverride != nullptr)
+    {
+        bool dataWrite;
+        ConcreteAttributePath path(aClusterInfo.mEndpointId, aClusterInfo.mClusterId, aClusterInfo.mFieldId);
+
+        ReturnErrorOnFailure(attrOverride->Write(path, aReader, &dataWrite));
+
+        if (dataWrite)
+        {
+            return apWriteHandler->AddStatus(attributePathParams, Protocols::InteractionModel::Status::Success);
```

Comment:
Sending just success is not a good error handling strategy - see my earlier post about this.

Reply:
If there is any write error it will return before this check from 

ReturnErrorOnFailure(attrOverride->Write(path, aReader, &dataWrite));

Reply:
Same as above.  This will be handled in follow-up PR together with read. #11663